### ISSUE TITLE
Incorrect error when using prototype property from static initializer

### ISSUE THREAD

**Issue by RyanCavanaugh** [Microsoft/TypeScript#14580]
Reduced repro from https://github.com/Microsoft/TypeScript/issues/12609#issuecomment-285727393
```ts
class Foo {
    static x = { j: Foo.prototype.bar }; // Error here, should be OK
    bar() { }
}
```
The static initializers execute after the class prototype is set up, so this should be OK

**Comment by evil-shrike** [comment]
Hi, is there any news on the bug?

It'd great to have it fixed till 2.3 release...

**Comment by mhegazy** [comment]
should be fixed in latest.
